File: 0113-路径总和2/Solution.java
import java.util.ArrayList;
import java.util.List;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public List<List<Integer>> pathSum(TreeNode root, int sum) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null){ return res; }
        if(root.val == sum && root.left == null && root.right == null){ List<Integer> temp=new ArrayList<>(); temp.add(sum); res.add(temp); return res;}
        if(root.val != sum && root.left == null && root.right == null){ return res; }
        int newsum = sum - root.val;
        var left = pathSum(root.left, newsum); 
        var right = pathSum(root.right, newsum);
        for (int i = 0; i < left.size(); i++) {
            var curr = left.get(i);
            curr.add(0, root.val);
            res.add(curr);            
        }
        for (int i = 0; i < right.size(); i++) {
            var curr = right.get(i);
            curr.add(0, root.val);
            res.add(curr);
        }
        return res;
    }
}

File: 0113-路径总和2/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def pathSum(self, root: TreeNode, sum: int) -> List[List[int]]:
        if not root:
            return []
        ret = []
        self.dfs(root, sum, [], ret)
        return ret

    def dfs(self, root, target, path, ret):
        if root.left is None and root.right is None:
            if target == root.val:
                ret.append(path+[root.val])
            return None
        if root.left:
            self.dfs(root.left, target-root.val, path+[root.val], ret)
        if root.right:
            self.dfs(root.right, target-root.val, path+[root.val], ret)
        return None